src/jit/codegenarm64.cpp

```diff
@@ -2157,12 +2157,10 @@ void                CodeGen::instGen_Set_Reg_To_Imm(emitAttr    size,
     {
         size = EA_SIZE(size);  // Strip any Reloc flags from size if we aren't doing relocs
     }
-    
+
     if (EA_IS_RELOC(size))
     {
-        // Emit a data section constant for a relocatable integer constant.
-        CORINFO_FIELD_HANDLE hnd = getEmitter()->emitLiteralConst(imm);
-        getEmitter()->emitIns_R_C(INS_ldr, size, reg, hnd, 0);    
+        getEmitter()->emitIns_R_AI(INS_adrp, size, reg, imm);
```

Comment:
Do we generate two instructions here?
Add a comment explaining how that occurs.